stackrox/stackrox
# ROX-14398: Group Role resource with Access - migrate permission sets

## Description

This change is part of a series with the goal of making the configuration of user access control easier, and of limiting the risk of features not working because of missing permissions.

The goal is to bring the `Role` permission under the umbrella of the `Access` one.
This part ensures that the existing permission sets are updated to group `Role` and `Access` together under the `Access` permission.

## Checklist
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should cover the base

Manual testing:
- Spawn a 4.0.0 Central
- Create from the UI the 9 roles combining the `None`, `Read` and `ReadWrite` access levels for the permissions `Access` and `Role`, while leaving the default access level for the other permissions.
- Create from the API the 16 roles combining the `Absent`, `None`, `Read` and `ReadWrite` access levels for the permissions `Access` and `Role`, while leaving a set of permissions with given access levels.
- Upgrade Central to the latest pushed PR images.
- Check the migration ran (180 to 181) in Central logs
- Check from the API that the migrated permission sets do not have the `Role` permission anymore, and that the access level is the minimum of the initial `Access` and `Role` access levels.
